build_runner_core/lib/src/generate/build_impl.dart

```diff
@@ -424,13 +426,12 @@ class _SingleBuild {
         .putIfAbsent(phaseNumber, () => new Set<String>())
         .add(actionDescription);
 
-    var errorThrown = false;
     await tracker.track(
         () => runBuilder(builder, [input], wrappedReader, wrappedWriter,
                 new PerformanceTrackingResolvers(_resolvers, tracker),
                 logger: logger,
                 resourceManager: _resourceManager).catchError((_) {
-              errorThrown = true;
+              // Errors tracked throug logger
```

Comment:
Do we need to clean up the previous errors before running this builder again? See the _cleanupStaleOutputs call on line 413 above

Reply:
oh yeah, I had put in `FailureReporter.clean` but forgot to call it. Calling it from within `_cleanUpStaleOutputs` now.